src/test/java/com/github/emilshina/testcases/DataSuppliers.java

```diff
@@ -0,0 +1,44 @@
+package com.github.emilshina.testcases;
+
+import com.github.emilshina.data.DataReader;
+import com.github.emilshina.model.User;
+import com.github.emilshina.utils.ServiceLoaderUtils;
+import io.github.sskorol.core.DataSupplier;
+import lombok.extern.slf4j.Slf4j;
+import one.util.streamex.StreamEx;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Data suppliers' methods for tests.
+ */
+
+@Slf4j
+public class DataSuppliers {
+
+    @DataSupplier
+    public User usersWithInvalidCredentials(final Method method) {
+        final String dataTypeFromAnnotation = method.getDeclaredAnnotation(Data.class).dataType();
+        final String source = method.getDeclaredAnnotation(Data.class).source();
+        return StreamEx.of(ServiceLoaderUtils.load(DataReader.class, ClassLoader.getSystemClassLoader()))
+                .findFirst(dataReader -> dataReader.dataType().equals(dataTypeFromAnnotation))
+                .map(dataReader -> dataReader.readFrom(source, User.class))
```

Comment:
try passing entity class within `Data` annotation and make a generic data supplier